### PR TITLE
Static files handler

### PR DESCRIPTION
**Pull request by jasony0206**

### CODE REVIEW

File: handler_factory.cc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include <memory>
+#include "echo_handler.h"
+#include "static_file_handler.h"
+
+HandlerFactory::HandlerFactory(ServerConfig* server_config, HttpRequest* http_request)
+    : server_config_(server_config),
+      http_request_(http_request) {}
+
+std::unique_ptr<RequestHandler> HandlerFactory::create_handler() {
+    std::cout << "in create handler!" << std::endl;
```
Comment (tincan24) [inline handler_factory.cc]: not sure if the couts were used for debugging or actual logging

File: handler_factory_test.cc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#include "gtest/gtest.h"
+#include "handler_factory.h"
+
```
Comment (tincan24) [inline handler_factory_test.cc]: Empty test file, should add some test cases

File: static_file_handler_test.cc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+TEST_F(StaticFileHandlerTest, GetMimeTypeTestHTML) {
+    EXPECT_EQ("text/html", get_mime_type("bird.html"));
+}
+
+TEST_F(StaticFileHandlerTest, GetMimeTypeTestDefault) {
+    EXPECT_EQ("text/plain", get_mime_type("turtle"));
+}
+
+// TODO: Test read_file and build_response
```
Comment (tincan24) [inline static_file_handler_test.cc]: Should finish this TODO before merging

File: static_file_handler.h [hunk 4]
```diff
@@ -1,12 +1,20 @@
 #ifndef STATIC_FILE_HANDLER_H
 #define STATIC_FILE_HANDLER_H
+
+#include <boost/filesystem.hpp>
```
Comment (alexmwaz) [inline static_file_handler.h]: unnecessary includes now that we're only using ifstream i think

Comment (alexmwaz) [inline static_file_handler.h]: unnecessary includes now that we're only using ifstream i think

File: handler_factory.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+#include "request_handler.h"
+
+class HandlerFactory {
+public:
+    HandlerFactory(ServerConfig* server_config, HttpRequest* http_request);
+    std::unique_ptr<RequestHandler> create_handler();
+    std::string transform_path();
+    bool is_prefix(std::string short_str, std::string long_str);
+private:
+    ServerConfig* server_config_;
```
Comment (alexmwaz) [inline handler_factory.h]: are these two pointers deleted somewhere? possible memory leak

Comment (alexmwaz) [inline handler_factory.h]: wait never mind LOL

**Pull request merged**